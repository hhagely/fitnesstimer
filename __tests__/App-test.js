import 'react-native';
import React from 'react';
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<App />', () => {
	it('renders the loading screen', async () => {
		const tree = shallow(<App />);
		expect(toJson(tree)).toMatchSnapshot();
	});

	it('renders the root without loading screen', async () => {
		const tree = shallow(<App skipLoadingScreen />);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
