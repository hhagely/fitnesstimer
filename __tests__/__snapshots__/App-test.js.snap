// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the loading screen 1`] = `<ExponentAppLoading />`;

exports[`renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={undefined}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#EFEFF4",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            loaded={
              Array [
                0,
              ]
            }
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                testID={undefined}
              >
                <View
                  collapsable={false}
                  removeClippedSubviews={false}
                  style={
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "flex": 1,
                        }
                      }
                    >
                      <RCTScrollView
                        contentContainerStyle={
                          Object {
                            "flex": 1,
                            "paddingTop": 30,
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              Object {
                                "backgroundColor": "#fff",
                                "flex": 1,
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <View>
                              <View>
                                <Modal
                                  animationType="slide"
                                  hardwareAccelerated={false}
                                  onRequestClose={[Function]}
                                  transparent={false}
                                  visible={false}
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#fff",
                                          "flex": 1,
                                          "flexDirection": "column",
                                          "justifyContent": "space-around",
                                        },
                                        Object {
                                          "alignItems": "center",
                                        },
                                        false,
                                      ]
                                    }
                                  >
                                    <View>
                                      <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                          Object {
                                            "fontSize": 32,
                                          }
                                        }
                                      >
                                        Your workout will begin in:
                                      </Text>
                                    </View>
                                    <View>
                                      <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                          Object {
                                            "fontSize": 72,
                                          }
                                        }
                                      />
                                    </View>
                                    <View
                                      style={
                                        Object {
                                          "width": 125,
                                        }
                                      }
                                    >
                                      <View
                                        accessibilityComponentType="button"
                                        accessibilityLabel={undefined}
                                        accessibilityTraits={
                                          Array [
                                            "button",
                                          ]
                                        }
                                        accessible={true}
                                        collapsable={undefined}
                                        hasTVPreferredFocus={undefined}
                                        hitSlop={undefined}
                                        isTVSelectable={true}
                                        nativeID={undefined}
                                        onLayout={undefined}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Object {
                                            "opacity": 1,
                                          }
                                        }
                                        testID={undefined}
                                        tvParallaxProperties={undefined}
                                      >
                                        <View
                                          style={
                                            Array [
                                              Object {},
                                            ]
                                          }
                                        >
                                          <Text
                                            accessible={true}
                                            allowFontScaling={true}
                                            disabled={undefined}
                                            ellipsizeMode="tail"
                                            style={
                                              Array [
                                                Object {
                                                  "color": "#007AFF",
                                                  "fontSize": 18,
                                                  "padding": 8,
                                                  "textAlign": "center",
                                                },
                                              ]
                                            }
                                          >
                                            Cancel
                                          </Text>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </Modal>
                              </View>
                              <View>
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  ellipsizeMode="tail"
                                  style={
                                    Object {
                                      "fontSize": 108,
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  10
                                  :
                                  00
                                </Text>
                              </View>
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "space-around",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "width": 125,
                                    }
                                  }
                                >
                                  <View
                                    accessibilityComponentType="button"
                                    accessibilityLabel={undefined}
                                    accessibilityTraits={
                                      Array [
                                        "button",
                                      ]
                                    }
                                    accessible={true}
                                    collapsable={undefined}
                                    hasTVPreferredFocus={undefined}
                                    hitSlop={undefined}
                                    isTVSelectable={true}
                                    nativeID={undefined}
                                    onLayout={undefined}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "opacity": 1,
                                      }
                                    }
                                    testID={undefined}
                                    tvParallaxProperties={undefined}
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {},
                                        ]
                                      }
                                    >
                                      <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        disabled={undefined}
                                        ellipsizeMode="tail"
                                        style={
                                          Array [
                                            Object {
                                              "color": "#007AFF",
                                              "fontSize": 18,
                                              "padding": 8,
                                              "textAlign": "center",
                                            },
                                          ]
                                        }
                                      >
                                        Start
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "width": 125,
                                    }
                                  }
                                >
                                  <View
                                    accessibilityComponentType="button"
                                    accessibilityLabel={undefined}
                                    accessibilityTraits={
                                      Array [
                                        "button",
                                        "disabled",
                                      ]
                                    }
                                    accessible={true}
                                    collapsable={undefined}
                                    hasTVPreferredFocus={undefined}
                                    hitSlop={undefined}
                                    isTVSelectable={true}
                                    nativeID={undefined}
                                    onLayout={undefined}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "opacity": 1,
                                      }
                                    }
                                    testID={undefined}
                                    tvParallaxProperties={undefined}
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {},
                                          Object {},
                                        ]
                                      }
                                    >
                                      <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        disabled={true}
                                        ellipsizeMode="tail"
                                        style={
                                          Array [
                                            Object {
                                              "color": "#007AFF",
                                              "fontSize": 18,
                                              "padding": 8,
                                              "textAlign": "center",
                                            },
                                            Object {
                                              "color": "#cdcdcd",
                                            },
                                          ]
                                        }
                                      >
                                        Reset
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={null}
                testID={undefined}
              />
              <View
                style={null}
                testID={undefined}
              />
            </View>
            <View
              collapsable={undefined}
              style={undefined}
            >
              <View
                collapsable={undefined}
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "backgroundColor": "#F7F7F7",
                    "borderTopColor": "rgba(0, 0, 0, .3)",
                    "borderTopWidth": 0.5,
                    "flexDirection": "row",
                    "height": 49,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, 0)",
                      "flex": 1,
                    }
                  }
                  testID={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        Object {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "height": 29,
                          },
                          false,
                          Object {
                            "flex": 1,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={undefined}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 30,
                            "opacity": 1,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#2f95dc",
                                "fontSize": 28,
                              },
                              Object {
                                "marginBottom": -3,
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                      <View
                        collapsable={undefined}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 30,
                            "opacity": 0,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#ccc",
                                "fontSize": 28,
                              },
                              Object {
                                "marginBottom": -3,
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      collapsable={undefined}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "rgba(52, 120, 246, 1)",
                          "fontSize": 10,
                          "marginBottom": 1.5,
                          "textAlign": "center",
                        }
                      }
                    >
                      Home
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, 0)",
                      "flex": 1,
                    }
                  }
                  testID={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        Object {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "height": 29,
                          },
                          false,
                          Object {
                            "flex": 1,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={undefined}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 30,
                            "opacity": 0,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#2f95dc",
                                "fontSize": 28,
                              },
                              Object {
                                "marginBottom": -3,
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                      <View
                        collapsable={undefined}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 30,
                            "opacity": 1,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#ccc",
                                "fontSize": 28,
                              },
                              Object {
                                "marginBottom": -3,
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      collapsable={undefined}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "rgba(146, 146, 146, 1)",
                          "fontSize": 10,
                          "marginBottom": 1.5,
                          "textAlign": "center",
                        }
                      }
                    >
                      Links
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, 0)",
                      "flex": 1,
                    }
                  }
                  testID={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        Object {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "height": 29,
                          },
                          false,
                          Object {
                            "flex": 1,
                          },
                        ]
                      }
                    >
                      <View
                        collapsable={undefined}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 30,
                            "opacity": 0,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#2f95dc",
                                "fontSize": 28,
                              },
                              Object {
                                "marginBottom": -3,
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                      <View
                        collapsable={undefined}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 30,
                            "opacity": 1,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#ccc",
                                "fontSize": 28,
                              },
                              Object {
                                "marginBottom": -3,
                              },
                              Object {
                                "fontFamily": "ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      collapsable={undefined}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "rgba(146, 146, 146, 1)",
                          "fontSize": 10,
                          "marginBottom": 1.5,
                          "textAlign": "center",
                        }
                      }
                    >
                      Timer Settings
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
